Sources/Mantis/CropViewController/CropViewController.swift

```diff
@@ -378,9 +422,42 @@ private extension CropViewController {
 extension CropViewController: CropViewDelegate {    
     func cropViewDidBecomeResettable(_ cropView: CropViewProtocol) {
         cropToolbar.handleCropViewDidBecomeResettable()
-        delegate?.cropViewControllerDidImageTransformed(self)
         delegate?.cropViewControllerDidImageTransformed(self, transformation: cropView.makeTransformation())
         delegate?.cropViewController(self, didBecomeResettable: true)
+        
+        if config.enableUndoRedo {
+            guard let previous = previousCropState else { return }
+            let userGenerated = isCropStateUserGenerated
+            currentCropState = cropView.makeCropState()
+            
+            if previous != currentCropState {
+                pushTransformRecordOntoStack(transformType: .transform, previous: previous, current: currentCropState, userGenerated: userGenerated)
+            }
+            
+            previousCropState = nil
+            currentCropState = nil
+        }
+    }
+    
+    private func pushTransformRecordOntoStack(transformType: TransformType, previous: CropState, current: CropState, userGenerated: Bool) {
+        if userGenerated {
+            
+            let actionString: String!
+            
+            switch transformType {
+            case .transform:
+                actionString = NSLocalizedString("Change Crop", comment: "Change Crop")
+            case .resetTransforms:
+                actionString = NSLocalizedString("Reset Changes", comment: "Reset Changes")
+            }
+            
+            let previousValue :  [String : Any?] = [.kCurrentTransformState : previous]
+            let currentValue :  [String : Any?] = [.kCurrentTransformState : current]
+            
```

Comment:
remove space in front of `:` for `previousValue` and `currentValue`

Reply:
change has been made.

Reply:
We also need to remove extra space for `[String : Any?]` and `[.kCurrentTransformState : previous]` to keep consistent

Reply:
removed